File: Data Structures & Algorithms/combinations-of-a-phone-number/submission-4.py
class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        lDict = {
            "2": "abc",
            "3": "def",
            "4": "ghi",
            "5": "jkl",
            "6": "mno",
            "7": "pqrs",
            "8": "tuv",
            "9": "wxyz",
        }

        results = []

        def dfs(index, word):
            if len(word) == len(digits):
                results.append(word)
                return

            for ch in lDict.get(digits[index]):
                dfs(index + 1, word + ch)

        if len(digits) > 0:
            dfs(0, "")

        return results

File: Data Structures & Algorithms/longest-common-prefix/submission-4.cpp
class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        for(int index=0; index < strs[0].size(); index++){
            for( auto&s : strs){
                if(s.size()-1 < index || s[index]!=strs[0][index]){
                    return s.substr(0,index);
                }
            }
        }
        return strs[0];
    }
};